machine/accton/accton_as_28t_ops/kernel/platform-as_28t_ops.patch

```diff
@@ -0,0 +1,479 @@
++	.dt_compat  = hurricane2_dt_board_compat,
++MACHINE_END
++
++static const char * helix4_dt_board_compat[] = {
++    NULL
++};
++
+ DT_MACHINE_START(HELIX4_DT, "Broadcom Helix4 (Flattened Device Tree)")
+ 	.map_io     = board_map_io,
+ 	.init_early = iproc_init_early,
+diff --git a/arch/arm/mach-iproc/flash.c b/arch/arm/mach-iproc/flash.c
+index 5eaf3ec..dec9d70 100644
+--- a/arch/arm/mach-iproc/flash.c
++++ b/arch/arm/mach-iproc/flash.c
+@@ -44,22 +44,32 @@
+ #if defined(CONFIG_IPROC_QSPI) || defined(CONFIG_IPROC_QSPI_MODULE)
+ static struct mtd_partition sflash_partition_map[] = {
+     {
+-        .name = "boot",
++        .name = "uboot",
+         .offset = 0x00000000,
+-        .size = 640 * 1024,
++        .size = 832 * 1024,
+     },
+     {
+-        .name = "env",
++        .name = "shmoo",
+         .offset = MTDPART_OFS_APPEND,
+-        .size = 384 * 1024,
++        .size = 64 * 1024,
+     },
+     {
+-        .name = "system",
++        .name = "uboot-env",
+         .offset = MTDPART_OFS_APPEND,
+-        .size = 15 * 1024 * 1024,
++        .size = 64 * 1024,
+     },
+     {
+-        .name = "rootfs",
++        .name = "onie",
++        .offset = MTDPART_OFS_APPEND,
++        .size = 7 * 1024 * 1024,
++    },
++    {
++        .name = "hw-info",
++        .offset = MTDPART_OFS_APPEND,
++        .size = 64 * 1024,
++    },
++    {
++        .name = "diag",
+         .offset = MTDPART_OFS_APPEND,
+         .size = MTDPART_SIZ_FULL,
+     },
+@@ -69,28 +79,8 @@ static struct mtd_partition sflash_partition_map[] = {
+ #if defined(CONFIG_IPROC_MTD_NAND) || defined(CONFIG_IPROC_MTD_NAND_MODULE)
+ static struct mtd_partition nand_partition_map[] = {
+     {
+-        .name = "nboot",
++        .name = "open",
+         .offset = 0x00000000,
+-        .size = 2 * 1024 * 1024,
+-    },
+-    {
+-        .name = "nenv",
+-        .offset = MTDPART_OFS_APPEND,
+-        .size = 4 * 1024 * 1024,
+-    },
+-    {
+-        .name = "nsystem",
+-        .offset = MTDPART_OFS_APPEND,
+-        .size = 10 * 1024 * 1024,
+-    },
+-    {
+-        .name = "nrootfs",
+-        .offset = MTDPART_OFS_APPEND,
+-        .size = 48 * 1024 * 1024,
+-    },
+-    {
+-        .name = "ncustfs",
+-        .offset = MTDPART_OFS_APPEND,
+         .size = MTDPART_SIZ_FULL,
+     },
+ };
+diff --git a/drivers/bcmdrivers/gmac/src/et/sys/etcgmac.c b/drivers/bcmdrivers/gmac/src/et/sys/etcgmac.c
+index cf16543..071c991 100755
+--- a/drivers/bcmdrivers/gmac/src/et/sys/etcgmac.c
++++ b/drivers/bcmdrivers/gmac/src/et/sys/etcgmac.c
+@@ -428,6 +428,7 @@ chipattach(etc_info_t *etc, void *osh, void *regsva)
+ 	} else {
+ 		etc->phyaddr = bcm_atoi(var) & EPHY_MASK;
+ 	}
++	etc->phyaddr = 0x18;
+ 	printf("et%d: chipattach: phyaddr(0x%x)\n", etc->unit, etc->phyaddr);
+ 
+ 	/* nvram says no phy is present */
+@@ -2273,7 +2274,7 @@ chipphyrd(ch_t *ch, uint phyaddr, uint reg)
+ 	uint32 addr, bank;
+ 	uint16 tmp16;
+ 
+-	addr = phyaddr&0xf;
++	addr = phyaddr&0x1f;
```

Comment:
The PHY of mgmt port locates at 0x18.  To extract the correct address, the bit mask has to be revised 0x1f.